src/opera/pge/dist_s1/dist_s1_pge.py

```diff
@@ -31,12 +33,226 @@ class DistS1PreProcessorMixin(PreProcessorMixin):
 
     This particular pre-processor is currently a stub implementation, inheriting from the base pre-processor mixin
     and adding nothing at this time. New functionalities will be added as new versions of the DIST-S1 SAS are released.
-
     """
 
     _pre_mixin_name = "DistS1PreProcessorMixin"
     _valid_input_extensions = (".tif",)
 
+    _rtc_pattern = re.compile(r'(?P<id>(?P<project>OPERA)_(?P<level>L2)_(?P<product_type>RTC)-(?P<source>S1)_'
+                              r'(?P<burst_id>\w{4}-\w{6}-\w{3})_(?P<acquisition_ts>\d{4}\d{2}\d{2}T\d{2}\d{2}\d{2}Z)'
+                              r'_(?P<creation_ts>\d{4}\d{2}\d{2}T\d{2}\d{2}\d{2}Z)_(?P<sensor>S1A|S1B|S1C)_'
+                              r'(?P<spacing>30)_(?P<product_version>v\d+[.]\d+))'
+                              r'(_(?P<pol>VV|VH|HH|HV|VV\+VH|HH\+HV))?[.](?P<ext>tif|tiff)$')
+
+    def __validate_rtc_list_lengths(self):
+        """
+        Validate that the pre_rtc_copol & pre_rtc_crosspol / post_rtc_copol &
+        post_rtc_crosspol are the same lengths
+        """
+        sas_config = self.runconfig.sas_config
+
+        pre_rtc_copol = sas_config["run_config"]["pre_rtc_copol"]
+        pre_rtc_crosspol = sas_config["run_config"]["pre_rtc_crosspol"]
+        post_rtc_copol = sas_config["run_config"]["post_rtc_copol"]
+        post_rtc_crosspol = sas_config["run_config"]["post_rtc_crosspol"]
+
+        if len(pre_rtc_copol) != len(pre_rtc_crosspol) or len(post_rtc_copol) != len(post_rtc_crosspol):
+            msg = (f'Lengths of input pre/post co/cross pol input RTC lists differ: pre {len(pre_rtc_copol)} '
+                   f'{len(pre_rtc_crosspol)} post {len(post_rtc_copol)} {len(post_rtc_crosspol)}')
+            self.logger.critical(
+                self.name,
+                ErrorCode.INVALID_INPUT,
+                msg
+            )
+
+    def __validate_rtc_lists_are_pge_subset(self):
+        """Ensures all RTCs in the SAS group are also in the PGE group"""
+        sas_config = self.runconfig.sas_config
+
+        pre_rtc_copol = sas_config["run_config"]["pre_rtc_copol"]
+        pre_rtc_crosspol = sas_config["run_config"]["pre_rtc_crosspol"]
+        post_rtc_copol = sas_config["run_config"]["post_rtc_copol"]
+        post_rtc_crosspol = sas_config["run_config"]["post_rtc_crosspol"]
+
+        all_rtcs = pre_rtc_copol + pre_rtc_crosspol + post_rtc_copol + post_rtc_crosspol
+
+        if not set([basename(rtc) for rtc in all_rtcs]).issubset([basename(rtc) for rtc in self.runconfig.input_files]):
+            msg = 'SAS RTC file groups do not make a subset of PGE Input Files'
```

Comment:
Another instance where "RunConfig" should be used in place of "SAS". If these errors are ever flagged, the operators will be the first to see theses messages so I want to make sure we're extra clear where the error is.